### PR TITLE
Port mdis to kernel 4.15

### PR DESCRIPTION
**Pull request by yxude**
Replace pci_enable_msi_block with pci_enable_msix_exact
Fix incompatible casting
Replace init_timer with timer_setup
Use <linux/uaccess.h> instead of <asm/uaccess.h>
Do not print __DATE__ and __TIME__
Use type wait_queue_entry_t instead of wait_queue_t
Unify definition of DmaSetup
Remove duplicated definition: debug = DEBUG_DEFAULT
Function check_mem_region is dropped in new kernel
Unify definition of Tsi148_DmaSetup
Do not print GIT_VERSION

Signed-off-by: Xu, Yi <Yi.Xu@men.de>

### CODE REVIEW

File: DRIVERS/ETH_16Z077/DRIVER/men_16z077_eth.c [hunk 1]
```diff
@@ -2414,10 +2414,8 @@ int men_16z077_probe( CHAMELEON_UNIT_T *chu )
 	}
 
 	/* set up timer to poll for link state changes */
-	init_timer(&np->timer);
 	np->timer.expires 	= jiffies + CONFIG_HZ / 2;
-	np->timer.data 		= (unsigned long)dev;
-	np->timer.function 	= z77_timerfunc;
+	timer_setup(&np->timer, (void *)(struct timer_list *)z77_timerfunc, 0);
```
Comment (awujek) [inline DRIVERS/ETH_16Z077/DRIVER/men_16z077_eth.c]: timer_setup is not present in the kernel 3.6.11

File: TOOLS/VME4L_API/VME4L_RWEX/vme4l_rwex.c [hunk 2]
```diff
@@ -118,7 +118,7 @@ static void usage(int excode)
 	printf("              if fails, store read data in <file>"VERIFY_FILE_POSTFIX"\n");
 	printf("-m            memory map instead of using ioctl calls\n");
 	printf("\n");
-	printf("git version: %s\n", GIT_VERSION);
```
Comment (awujek) [inline TOOLS/VME4L_API/VME4L_RWEX/vme4l_rwex.c]: I'm against this change

File: LIBSRC/OSS/oss_sem.c [hunk 3]
```diff
@@ -237,7 +237,7 @@ int32 OSS_SemWait(
 
 	/* sem->lock is locked here */
 	{
-		wait_queue_t __wait;
+		wait_queue_entry_t __wait;
```
Comment (awujek) [inline LIBSRC/OSS/oss_sem.c]: not present in the kernel 3.6.11

File: DRIVERS/VME_16Z002/vme4l-core.h [hunk 4]
```diff
@@ -149,6 +149,7 @@ typedef void *VME4L_BRIDGE_HANDLE;
 #else
 #define DEBUG_DEFAULT 0
 #endif
+static int debug = DEBUG_DEFAULT;  /**< enable debug printouts */
```
Comment (awujek) [inline DRIVERS/VME_16Z002/vme4l-core.h]: I'm strongly against this. What if a file including vme4l-core.h has a debug variable for something else?

**Review by awujek — CHANGES_REQUESTED** [review]
I'm against about most of the changes.
Please keep in mind that this code shall be working for the kernel 3.6.11
Additionally, It is not clear to me why files from the  submodule DRIVERS/BBIS/D203, were added directly to the repo.

File: DRIVERS/ETH_16Z077/DRIVER/men_16z077_eth.c (hunk 1 above)
Comment (yxude) [reply DRIVERS/ETH_16Z077/DRIVER/men_16z077_eth.c]: Thanks for the hint. I'll let it apply for kernel version bigger than 4.15.

File: TOOLS/VME4L_API/VME4L_RWEX/vme4l_rwex.c (hunk 2 above)
Comment (yxude) [reply TOOLS/VME4L_API/VME4L_RWEX/vme4l_rwex.c]: I didn't find out where GIT_VERSION is defined, and got compile errors if not commented out.

File: LIBSRC/OSS/oss_sem.c (hunk 3 above)
Comment (yxude) [reply LIBSRC/OSS/oss_sem.c]: I'll let the change apply for kernel version bigger than 4.13.

File: TOOLS/VME4L_API/VME4L_RWEX/vme4l_rwex.c (hunk 2 above)
Comment (awujek) [reply TOOLS/VME4L_API/VME4L_RWEX/vme4l_rwex.c]: It should come from the build system. Please change the build system accordingly.

File: DRIVERS/VME_16Z002/vme4l-core.h (hunk 4 above)
Comment (yxude) [reply DRIVERS/VME_16Z002/vme4l-core.h]: In all vme4l*c files there's debug = DEBUG_DEFAULT defined, so isn't it simpler to define it in vme4l-core.h?

Comment (awujek) [reply DRIVERS/VME_16Z002/vme4l-core.h]: no, because vme4l-core.h may (and actually is) included by different drivers

**Comment by yxude** [comment]
PR updated. Please review.

File: DRIVERS/VME_16Z002/vme4l-core.h [hunk 5]
```diff
@@ -150,6 +154,7 @@ typedef void *VME4L_BRIDGE_HANDLE;
 #define DEBUG_DEFAULT 0
 #endif
 
+static int debug;
```
Comment (awujek) [inline DRIVERS/VME_16Z002/vme4l-core.h]: Please remove it.

**Review by awujek — CHANGES_REQUESTED** [review]
please remove debug variable from .h file
What about GIT_VERSION? will you change it?

File: DRIVERS/VME_16Z002/vme4l-core.h (hunk 5 above)
Comment (yxude) [reply DRIVERS/VME_16Z002/vme4l-core.h]: But it will then not compile even with gcc 4.8, it will complain that debug is not defined.

**Comment by yxude** [comment]
I can leave GIT_VERSION as it is.

File: DRIVERS/VME_16Z002/vme4l-core.h (hunk 5 above)
Comment (awujek) [reply DRIVERS/VME_16Z002/vme4l-core.h]: I think you have to add "static int debug;" to  DRIVERS/VME_16Z002/vme4l-tsi148.c and DRIVERS/VME_16Z002/vme4l-pldz002.c

Comment (yxude) [reply DRIVERS/VME_16Z002/vme4l-core.h]: In those c files debug is defined: static int debug = DEBUG_DEFAULT;  The complain comes from vme4l-core.h header file, in line 156, there debug is used but not defined.

Comment (awujek) [reply DRIVERS/VME_16Z002/vme4l-core.h]: I agree that debug is defined in thevme4l-tsi148.c, but not in the vme4l-pldz002.c.
Could you paste me the exact error?

Comment (yxude) [reply DRIVERS/VME_16Z002/vme4l-core.h]: Sorry I overlooked it. Updated PR. Thanks.

**Comment by awujek** [comment]
Can you squash these two commits?

**Comment by yxude** [comment]
@awujek done.

**Review by ThomasSchnuerer — APPROVED** [review]

**Pull request merged**